custom_components/plugwise/__init__.py

```diff
@@ -48,161 +21,12 @@ async def async_setup(hass: HomeAssistant, config: dict):
-        update_interval=update_interval,
-    )
-
-    await coordinator.async_refresh()
-
-    if not coordinator.last_update_success:
-        raise ConfigEntryNotReady
-
-    _LOGGER.debug("Async update interval %s", update_interval)
-
-    api.get_all_devices()
-
-    undo_listener = entry.add_update_listener(_update_listener)
-
-    # Migrate to a valid unique_id when needed
-    if entry.unique_id is None:
-        if api.smile_version[0] != "1.8.0":
-            hass.config_entries.async_update_entry(entry, unique_id=api.smile_hostname)
-
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
-        API: api,
-        COORDINATOR: coordinator,
-        UNDO_UPDATE_LISTENER: undo_listener,
-    }
-
-    _LOGGER.debug("Gateway is %s", api.gateway_id)
-
-    _LOGGER.debug("Gateway sofware version is %s", api.smile_version)
-    _LOGGER.debug("Appliances is %s", api.get_all_appliances())
-    _LOGGER.debug("Scan thermostats is %s", api.scan_thermostats())
-    _LOGGER.debug("Locations (matched) is %s", api.match_locations())
-
-    device_registry = await dr.async_get_registry(hass)
-    device_registry.async_get_or_create(
-        config_entry_id=entry.entry_id,
-        identifiers={(DOMAIN, api.gateway_id)},
-        manufacturer="Plugwise",
-        name=entry.title,
-        model=f"Smile {api.smile_name}",
-        sw_version=api.smile_version[0],
-    )
-
-    single_master_thermostat = api.single_master_thermostat()
-    _LOGGER.debug("Single master thermostat = %s", single_master_thermostat)
-
-    platforms = ALL_PLATFORMS
-    if single_master_thermostat is None:
-        platforms = SENSOR_PLATFORMS
-
-    async def async_delete_notification(self):
-        """Service: delete the Plugwise Notification."""
-        _LOGGER.debug("Service delete PW Notification called for %s", api.smile_name)
-        try:
-            deleted = await api.delete_notification()
-            _LOGGER.debug("PW Notification deleted: %s", deleted)
-        except Smile.PlugwiseError:
-            _LOGGER.debug(
-                "Failed to delete the Plugwise Notification for %s", api.smile_name
-            )
-
-    for component in platforms:
-        hass.async_create_task(
-            hass.config_entries.async_forward_entry_setup(entry, component)
-        )
-        if component == "climate":
-            hass.services.async_register(
-                DOMAIN, SERVICE_DELETE, async_delete_notification, schema=vol.Schema({})
-            )
-
-    return True
-
-
-async def _update_listener(hass: HomeAssistant, entry: ConfigEntry):
-    """Handle options update."""
-    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
-    coordinator.update_interval = timedelta(
-        seconds=entry.options.get(CONF_SCAN_INTERVAL)
-    )
-
+    """Set up Plugwise components from a config entry."""
+    if entry.data.get(CONF_HOST):
+        return await async_setup_entry_gw(hass, entry)
+    # PLACEHOLDER USB entry setup
```

Comment:
True if we don't want to include #117 just yet (which was the goal, right)

Reply:
Yes, that's the idea, first, implement the structure-change from Core. Next step, implement #117.